### ISSUE TITLE
Three test cases of FiberCacheManagerSuite failed unexpectedly.

### ISSUE THREAD

**Issue by lee-lei** [Intel-bigdata/OAP#750]
Three test cases of FiberCacheManagerSuite are below:
"cache guardian remove pending fibers"
"get different fiber simultaneously"
"LRU blocks memory free"

The above three cases sometimes failed on github, although sometimes
it passed when individually running locally.

The reason is that the fiber caches created in the above cases are
not successfully removed from the fiber cache manager in the
predefied duration in the cases.

In other words, the synchronization between fiber creation thread (current user thread)
and the fiber cache manager thread is not robust enough. Thus sometimes the fiber cache
manager will delay to truly release the unused fiber caches a long while after the fiber notified the fiber cache manager to release itself.  

The root cause will be analyzed accordingly.

**Comment by lee-lei** [comment]
Problem1: The fibers are cached in the fiber cache manager. After the fiber cache manager removed the fiber, this evicted fiber will be filled into another data structure LinkedBlockingQueue in order to synchronize between the user main thread and CacheGuardian thread. This redundant second copy increases the memory consumption.

**Comment by lee-lei** [comment]
Problem3: The guava cache manager only support least recently used (LRU) eviction algorithm. The least frequently used (LFU) replacemnt is expected to have better cache performance than LRU for the big data use cases. However, the LFU is not supported in the guava cache manager.

**Comment by linhong-intel** [comment]
Thanks @lee-lei 
Small question: any idea about why the 3 test case failed on travis?

**Comment by adrian-wang** [comment]
Even run tests locally, the 3 test cases could also fail by chance.

**Comment by carsonwang** [comment]
@lee-lei , can you create separated issues for the problems identified?

**Comment by LuciferYang** [comment]
We need to prioritize fix this problem, sometimes travis failed because of these cases
